File: common/src/main/java/com/github/lukebemish/clojuremixin/platform/Platform.java
package com.github.lukebemish.clojuremixin.platform;

import dev.architectury.injectables.annotations.ExpectPlatform;

public enum Platform {
    FORGE("forge"),
    FABRIC("fabric");

    private final String name;

    @ExpectPlatform
    public static Platform getPlatform() {
        throw new AssertionError();
    }

    public static Object unimplError() {
        throw new AssertionError();
    }

    Platform(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

File: common/src/main/clojure/com/github/lukebemish/clojuremixin/api/platform.clj
(ns com.github.lukebemish.clojuremixin.api.platform
  (:import (com.github.lukebemish.clojuremixin.platform Platform)))

(def forge Platform/FORGE)
(def fabric Platform/FABRIC)

(defn get-platform [] (Platform/getPlatform))
(defn unimpl-error [] (Platform/unimplError))

(defn fabric? [] (= (get-platform) fabric))
(defn forge? [] (= (get-platform) forge))